### PR TITLE
Exercises QA API

### PR DESCRIPTION
**Pull request by Dantemss**
- Modified ecosystem exercises route to return all exercises by default
- Added pool_types parameter to limit the types of exercises returned
- Added specs

### CODE REVIEW

**Comment by Dantemss** [comment]
Needs corresponding FE PR to changed the route used in the HW builder to `/ecosystems/:ecosystem_id/exercises/homework_core`

**Comment by Dantemss** [comment]
Includes https://github.com/openstax/tutor-server/pull/627.

File: app/controllers/api/v1/ecosystems_controller.rb [hunk 1]
```diff
@@ -49,15 +49,45 @@ def respond_with_ecosystem_readings(ecosystem)
     respond_with books, represent_with: Api::V1::BookTocsRepresenter
   end
 
-  def respond_with_ecosystem_exercises(ecosystem)
+  def respond_with_ecosystem_exercises(ecosystem, pool_types = nil)
```
Comment (jpslav) [inline app/controllers/api/v1/ecosystems_controller.rb]: Can most of the code in here go into a routine? (or somewhere else)  skinny controllers make me smile

File: app/controllers/api/v1/ecosystems_controller.rb [hunk 2]
```diff
@@ -49,15 +49,45 @@ def respond_with_ecosystem_readings(ecosystem)
...
     OSU::AccessPolicy.require_action_allowed!(:exercises, current_api_user, ecosystem)
 
     pages = ecosystem.pages_by_ids(params[:page_ids])
-    exercises = ecosystem.homework_core_pools(pages: pages).flat_map do |pool|
-      pool.exercises(preload_tags: true)
+
+    pool_types = [pool_types].flatten.compact
+
+    # Default types
+    pool_types = ['reading_dynamic', 'reading_try_another',
```
Comment (jpslav) [inline app/controllers/api/v1/ecosystems_controller.rb]: Is the default pool_type now the `all_exercises_pool`?

Comment (jpslav) [inline app/controllers/api/v1/ecosystems_controller.rb]: oh but I guess you're trying to list the pool_type for each exercise so need them broken down.  

Maybe there's some way to simplify this logic like...

```ruby
pools = {}
Content::Pool.distinct_pool_types.each do |pool_type|
  pools[pool_type] = ecosystem.pools(pages: pages, type: pool_type)
end
``` 

where `distinct_pool_types` includes all types except for `all_exercises`

**Comment by jpslav** [comment]
Reviewed

File: app/controllers/api/v1/ecosystems_controller.rb (hunk 2 above)
Comment (Dantemss) [inline app/controllers/api/v1/ecosystems_controller.rb]: No harm in adding the all_exercises_pool, maybe just some duplication.

File: app/controllers/api/v1/ecosystems_controller.rb (hunk 1 above)
Comment (Dantemss) [inline app/controllers/api/v1/ecosystems_controller.rb]: Sure, I guess.

**Comment by Dantemss** [comment]
Will need rebasing after https://github.com/openstax/tutor-server/pull/644 is merged.

**Comment by jpslav** [comment]
@openstax/tutor-fe-dev -- do y'all have a mating PR for this?

**Comment by nathanstitt** [comment]
@jpslav I think this is needed for the qa PR https://github.com/openstax/tutor-js/pull/747 so it can retrieve all the exercises for a given book_location

**Comment by jpslav** [comment]
Thanks @nathanstitt -- let me know when that is in.

**Comment by jpslav** [comment]
@Dantemss can you rebase?

**Comment by nathanstitt** [comment]
FE PR is completed and tested against this one, but not yet merged.

**Comment by jpslav** [comment]
Will review after scrum today.

**Pull request merged**